File: src/CandyHandyman.Application/DTOs/ChatDto.cs
using CandyHandyman.Core.Enums;

namespace CandyHandyman.Application.DTOs;

public class ConversationDto
{
    public Guid Id { get; set; }
    public UserDto? OtherUser { get; set; }
    public MessageDto? LastMessage { get; set; }
    public int UnreadCount { get; set; }
    public DateTime LastMessageAt { get; set; }
}

public class MessageDto
{
    public Guid Id { get; set; }
    public Guid SenderId { get; set; }
    public string SenderName { get; set; } = string.Empty;
    public string? SenderAvatar { get; set; }
    public string Content { get; set; } = string.Empty;
    public MessageType MessageType { get; set; }
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class SendMessageDto
{
    public Guid ReceiverId { get; set; }
    public string Content { get; set; } = string.Empty;
    public MessageType MessageType { get; set; } = MessageType.Text;
}

File: android/app/src/main/java/com/candy/handyman/data/remote/dto/ChatDto.kt
package com.candy.handyman.data.remote.dto

data class ConversationDto(
    val id: String,
    val otherUser: UserDto? = null,
    val lastMessage: MessageDto? = null,
    val unreadCount: Int = 0,
    val lastMessageAt: String
)

data class MessageDto(
    val id: String,
    val senderId: String,
    val senderName: String,
    val senderAvatar: String?,
    val content: String,
    val messageType: String,
    val isRead: Boolean,
    val createdAt: String
)

data class SendMessageDto(
    val receiverId: String,
    val content: String,
    val messageType: String = "Text"
)